test/Constraints/diagnostics.swift

```diff
@@ -281,7 +281,7 @@ func r18800223(_ i : Int) {
 
   
   var buttonTextColor: String?
-  _ = (buttonTextColor != nil) ? 42 : {$0}; // expected-error {{result values in '? :' expression have mismatching types 'Int' and '(_) -> _'}}
+  _ = (buttonTextColor != nil) ? 42 : {$0}; // expected-error {{type of expression is ambiguous without more context}}
```

Comment:
I agree, I think there might be something we could to about it because I'm not sure how useful is it to return `(_) -> _` either. Maybe we could add a special diagnostic to say something like `can't infer result type of right/left-hand side branch, other side is value of type '<Type>'', WDYT?